spark-scala/src/main/scala/com/matthewrathbone/spark/Main.scala

```diff
@@ -0,0 +1,40 @@
+package main.scala.com.matthewrathbone.spark
+
+import org.apache.spark.SparkConf
+import org.apache.spark.SparkContext
+import org.apache.spark.SparkContext._
+import org.apache.hadoop.mapreduce.Job
+import org.apache.hadoop.io.Text
+import org.apache.hadoop.mapred.TextOutputFormat
+import org.apache.spark.rdd.RDD
+import scala.collection.Map
+
+object Main {
+  def main(args: Array[String]) {
```

Comment:
Can we split this up so that you can pass the SparkContext in as a constructor argument? That way the code can be used in both production and in tests.

Something like this will allow you to test more of the main function by passing in paths to local tabular files that can be read with the test SparkContext

```scala

class ExampleJob(sc: SparkContext) {
  def run(users, transactions) = {} // reads data from text files and computes the results. This is what you test
}

object ExampleJob {
  def main(args: Array[String]) {
    val usersIn = args(0)
    val transactionsIn = args(1)
    val context = ... // real context
    val job = new ExampleJob(context)
    println(job.run(usersIn, transactionsIn).toString)
  }
}
```

Then testing you can use your test context:


```scala

val job = new ExampleJob(conf)
val result = job.run("./data/users.tsv", "./data/transactions.tsv")
// test here


```